### DISCUSSION TITLE
Server_Role API Key and Data Access

### DISCUSSION THREAD

**Discussion by BigDarkAkula** [supabase/supabase#12821] [General]
You guys just helped me figure out my recent issue on getting null values was related to RLS.  I am a bit confused, though, because the service role API key should bypass all of that, no?

I gave the server_role API key to my salesforce instance so it uses it when it pushes data to Supabase.  The problem is that the related trigger doesn't seem to think this key is eligible to pull data further down the line in a related function that is called.  

What exactly does the server_role key bypass and what else do I need to get setup to allow it to have full access to Supabase?  I really want to restrict these tables so that only my server_role key and myself can access them.  They aren't really meant for end users to ever be able to access.

Shouldn't I be able to basically have RLS enabled but no policies set and the service_role can blow right past it?

**Comment by kangmingtay** [comment]
@BigDarkAkula that's not how RLS works. When you have RLS enabled on a table, it follows the principle of least privilege. You can check out the Row-Level Security docs on postgres (https://www.postgresql.org/docs/current/ddl-rowsecurity.html) which provides alot of useful information on how RLS works.
